assets/src/apps/authoring/components/ComponentToolbar/AddComponentToolbar.tsx

```diff
@@ -111,9 +112,10 @@ const AddComponentToolbar: React.FC<{
     dispatch(setCurrentSelection({ selection: newPartData.id }));
 
     dispatch(setRightPanelActiveTab({ rightPanelActiveTab: RightPanelTabs.COMPONENT }));
+    dispatch(setCopiedPart({ copiedPart: null }));
   };
 
-  useKeyDown(handlePartPasteClick, ['KeyV'], { ctrlKey: true }, [copiedPart]);
+  useKeyDown(handlePartPasteClick, ['KeyV'], { ctrlKey: true }, [copiedPart,currentActivityTree]);
```

Comment:
If you copy a part and try it to paste on another screen using shortcuts, it wasn't working and was pasting the part on the same screen from where you copied it because `currentActivityTree` was outdated